src/main/java/eu/dissco/orchestration/backend/service/SourceSystemService.java

```diff
@@ -30,4 +38,20 @@ public SourceSystemRecord updateSourceSystem(String id, SourceSystem sourceSyste
     repository.updateSourceSystem(sourceSystemRecord);
     return sourceSystemRecord;
   }
+
+  public JsonApiWrapper getSourceSystemRecords(int pageNum, int pageSize, String path){
+    var ssRecords = repository.getSourceSystems(pageNum, pageSize);
+    var dataNode = wrapData(ssRecords);
+    return new JsonApiWrapper(dataNode, new JsonApiLinks(path));
+  }
+
+  List<JsonApiData> wrapData(List<SourceSystemRecord> ssRecords){
+    List<JsonApiData> dataNode = new ArrayList<>();
+
+    ssRecords.forEach(ss -> dataNode.add(new JsonApiData(ss.id(), HandleType.SOURCE_SYSTEM, mapper.valueToTree(ss))));
```

Comment:
Another option might be to stream this, 
ssRecords.stream().map().toList()
Then we wouldn't have to create an empty list first and then populate it